clients/src/test/java/org/apache/kafka/common/protocol/ErrorsTest.java

```diff
@@ -60,4 +61,14 @@ public void testNoneException() {
         assertNull("The NONE error should not have an exception", Errors.NONE.exception());
     }
 
+    @Test
+    public void testForExceptionInheritance() {
+        class ExtendedTimeoutException extends TimeoutException {}
```

Comment:
Instead of having this temporary class, can we actually put this test in core and test all the scala Exception class to make sure they are returning the same error code as corresponding client side exception?